### PR TITLE
Added Companies tests, fixed bugs, and refactored logo containers

### PR DESCRIPTION
**Pull request by ZacheryEC001**
This pull request seeks to address some of the bugs and constructive criticism of the Companies component, including resolving issue #132. 

## Files Changed

**Companies.jsx** 
* Refactored the logo containers to use flex instead of grid, allowing for logos to be centered when items are removed by the user
* Fixed a bug where user-inputted logos would disappear in dark mode
* Removed unused imports

## Files Added

**Companies.test.jsx**
* Added unit tests

## Screenshots
image
_User-inputted logos now work in dark mode_

image
_Thanks to the refactor, logos will now center themselves if the user chooses less than 12 logos_

### CODE REVIEW

**Review by carruthjhall — CHANGES_REQUESTED** [review]
+The responsive styling works as intended, and the component looks good in both light and dark modes.
+The options are able to be edited and the changes are reflected on the screen.
-The options themselves would look better if there was a padding added to the left side using `pl-4` or something around that range.
-Adding the functionality to upload an image rather using an URL might be more suitable. You can find an example of how to do this on the skills component (if it has not been merged, then check PR #126 ). 
-Maybe put the logo options inside of a drop down like the Skills List dropdown here:
image
image
This could be done like so: 
`<details>`
`    <summary>Company Logos<summary>`
`         <details>`
`               <summary>Logo {logo #} URL</summary>`
`                   Logo option code goes here `
`         </details>`  
`          ...repeat for rest of logos` 
`</details>`

Able to get coverage on all but 2 of the files. You can use the test file provided by the Banner component to get the coverage you need for CompaniesOptions.jsx.
image

Overall good work, just a couple more fixes and this component will be perfect

**Comment by Lettuce05** [comment]
Had to close and reopen pull request in order to run GitHub action. Action would not run on preexisting pull requests.

**Comment by ZacheryEC001** [comment]
> +The responsive styling works as intended, and the component looks good in both light and dark modes. +The options are able to be edited and the changes are reflected on the screen. -The options themselves would look better if there was a padding added to the left side using `pl-4` or something around that range. -Adding the functionality to upload an image rather using an URL might be more suitable. You can find an example of how to do this on the skills component (if it has not been merged, then check PR #126 ). -Maybe put the logo options inside of a drop down like the Skills List dropdown here: image image This could be done like so: `<details>` ` <summary>Company Logos<summary>` ` <details>` ` <summary>Logo {logo #} URL</summary>` ` Logo option code goes here` ` </details>` ` ...repeat for rest of logos` `</details>`
> 
> Able to get coverage on all but 2 of the files. You can use the test file provided by the Banner component to get the coverage you need for CompaniesOptions.jsx. image
> 
> Overall good work, just a couple more fixes and this component will be perfect

Thank you for the feedback, I have implemented the requested changes

File: src/components/Companies/Companies.test.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import CompaniesOptions from './CompaniesOptions';
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+//default test options
+let options = {
+    gradient: Gradients.default,
+        noOfLogos: 12,
+        logoList: [
+            {url: "/src/components/Companies/Logos/Logo1.svg", darkURL: "/src/components/Companies/Logos/DarkLogo1.svg"},
```
Comment (ebacolod) [inline src/components/Companies/Companies.test.jsx:13]: It was mentioned in Discord that utilizing a file path as the source of your images cannot render in production mode. The solution to this was to host the images on the resources repository (https://github.com/cs472-team6/cs472-team6-resources).

**Review by ebacolod — CHANGES_REQUESTED** [review]

File: src/components/Companies/CompaniesOptions.jsx [hunk 2]
```diff
@@ -14,26 +14,43 @@ export default function CompaniesOptions({options, updateComponent}){
...
 return(
-    <div>
-        <label className="block">Logo {index + 1} URL</label>
-        <input className="block mb-3 p-2 rounded-lg bg-gray-200" type="url" name={`url`} value={url} onChange={event => handleLogoChange(index, event)}></input>
-    </div>
+    <details className="bg-white mx-1 rounded-lg">
+        <summary className="mb-3 cursor-pointer">Logo {index + 1}</summary>
+            <label className="block">Upload Logo {index + 1}</label>
+            <input className="block mb-3" type="file" accept="image/png, image/jpeg, image/svg+xml" name="url" onChange={event => handleImageChange(index, event)}></input>
+            <p className="mt-2 ml-1 text-xs">PNG, JPG, or SVG</p>
```
Comment (carruthjhall) [inline src/components/Companies/CompaniesOptions.jsx:53]: Just to make the new dropdowns format nicely, if you want you could do:
summary className: Change `mb-3` to `mb-1`, and add `pl-4`
label, input, and p className: Add `pl-7`

File: src/components/Companies/Companies.jsx [hunk 3]
```diff
@@ -1,33 +1,73 @@
...
-                        <Company key = {index} index = {index} company={company}/>
-                    )
-                })}
+            <div className="">
+                <div className="flex align-center justify-center">
+                    {logoList.slice(0,noOfLogos > (screenDimensions.isSm ? 3 : 4) ? (screenDimensions.isSm ? 3 : 4)  : noOfLogos).map((company, index) => {
+                            return(
+                                <Company key = {index} index = {index} company={company}/>
+                            )
+                        })}
```
Comment (carruthjhall) [inline src/components/Companies/Companies.jsx:35]: Wrap this like
`<details>`
`<summary>Logos</summary>`
`  {logolistarraycomponent} `
`</details>`
to clean it up a bit. But this looks great and the functionality works perfect.

**Review by carruthjhall — CHANGES_REQUESTED** [review]
Great work! Just some minor changes and this will be ready for merging.

File: src/components/Companies/CompaniesOptions.jsx (hunk 2 above)
Comment (ZacheryEC001) [reply src/components/Companies/CompaniesOptions.jsx]: Fixed.

File: src/components/Companies/Companies.jsx (hunk 3 above)
Comment (ZacheryEC001) [reply src/components/Companies/Companies.jsx:35]: Fixed.

File: src/components/Companies/Companies.jsx [hunk 4]
```diff
@@ -1,33 +1,73 @@
...
             </div>
-            <div className="grid grid-cols-3 md:grid-cols-4 gap-7 content-center">
-            {logoList.slice(0,noOfLogos).map((company, index) => {
-                    //let indexNo = index;
-                    return(
-                        <Company key = {index} index = {index} company={company}/>
-                    )
-                })}
+            <div className="">
+                <div className="flex align-center justify-center">
```
[the remark this answers is not in the document]
Comment (ZacheryEC001) [reply src/components/Companies/Companies.jsx]: Fixed.

File: src/components/Companies/Companies.test.jsx (hunk 1 above)
Comment (ZacheryEC001) [reply src/components/Companies/Companies.test.jsx:13]: @ebacolod At the moment, I do not have write access to that repo. Can you invite me to the organization or merge my PR on that repo? Thank you.

Comment (ZacheryEC001) [reply src/components/Companies/Companies.test.jsx]: Fixed.

**Review by ebacolod — APPROVED** [review]
Your component looks great! The images look a lot better when centered. I see the source images have been replaced with the URLs from the resource repo. Running in production, the images appear. Well done!

**Review by carruthjhall — APPROVED** [review]
The fixes look great and the component works great. Great work!

**Pull request merged**